client/src/modules/stock/movements/modals/search.modal.js

```diff
@@ -8,46 +8,52 @@ SearchMovementsModalController.$inject = [
 
 function SearchMovementsModalController(data, Notify, Instance, Flux, $translate, Periods, Store, util, Stock) {
   var vm = this;
+  var lastViewFilters;
+  var lastDisplayValues;
+  var displayValues = {};
   var changes = new Store({ identifier : 'key' });
 
   var searchQueryOptions = [
-    'is_exit', 'depot_uuid', 'inventory_uuid', 'label', 'flux_id', 'dateFrom', 'dateTo',
+    'is_exit', 'depot_uuid', 'inventory_uuid', 'label', 'flux_id', 'dateFrom', 'dateTo', 'user_id',
   ];
 
   vm.filters = data;
 
   vm.searchQueries = {};
   vm.defaultQueries = {};
 
-  // displayValues will be an id:displayValue pair
-  var displayValues = {};  
-
   // load flux
   Flux.read()
-    .then(function (rows) {
-      vm.fluxes = rows.map(function (row) {
-        row.label = $translate.instant(row.label);
-        return row;
-      });
-    })
+    .then(handleFluxes)
     .catch(Notify.handleError);
 
+  function handleFluxes(rows) {
+    vm.fluxes = rows.map(handleFlux);
+  }
+
+  function handleFlux(row) {
+    row.label = $translate.instant(row.label);
+    return row;
+  }
+
   // default filter period - directly write to changes list
   vm.onSelectPeriod = function onSelectPeriod(period) {
     var periodFilters = Periods.processFilterChanges(period);
 
-    periodFilters.forEach(function (filterChange) {
+    periodFilters.forEach(function handlePeriodFilter(filterChange) {
       changes.post(filterChange);
     });
   };
 
-  var lastViewFilters = Stock.filter.movement.formatView().customFilters;
+  lastViewFilters = Stock.filter.movement.formatView().customFilters;
 
   // map key to last display value for lookup in loggedChange
-  var lastDisplayValues = lastViewFilters.reduce(function (object, filter) {
+  lastDisplayValues = lastViewFilters.reduce(handleLastViewFilter, {});
```

Comment:
Most other registries put this on the service - see [the Patients search modal](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/modules/invoices/registry/search.modal.js#L24) for example.  The method is `getDisplayValueMap()` on the FilterService.